src/state_machine.cpp

```diff
@@ -0,0 +1,240 @@
+#include "state_machine.h"
+#include "sd_sector_addresses.h"
+
+#include <list>
+
+void StateMachine::construct() {
```

Comment:
In my opinion this appears a bit chaotic. Might be nice to first of all create all states and only after that link them together. So start with all the `createState()` functions. And after all states have been created you can start with the `withLeft()`, `withRight()`, `upTo()`, ... 

Like this, I think it will be easier to keep a good overview also when adding states.

Reply:
Do it just like you did the submenus ;)